### PR TITLE
Use `getBooleanInput` to parse flags

### LINKED ISSUE #175: Use `getBooleanInput` to parse flags
Adding a default value to the existing flag inputs should make those inputs required and simplify the conditionals slightly.

image

Essentially, the proposed change is to replace the way flags like `skipPush` are parsed:

```diff
- const skipPush = core.getInput('skipPush');
+ const skipPush = core.getBooleanInput('skipPush', { required: true });
```

### PR DESCRIPTION
**Pull request by sestrella**
Closes #175

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -21,9 +22,10 @@ inputs:
     description: 'Extra command-line arguments to pass to cachix. If empty, defaults to -j8'
   skipAddingSubstituter:
     description: 'Set to true to skip adding cachix cache as a substitute'
+    default: 'false'
   useDaemon:
     description: "Push store paths to the cache as they're built with the Cachix Daemon"
-    default: true
+    default: 'true'
```
Comment (sestrella) [inline action.yml:28]: According to the GH Actions documentation, the default value of an input should be specified as a string:

image


Reference: https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#inputsinput_iddefault

File: src/main.ts [hunk 2]
```diff
@@ -14,12 +14,12 @@ const name = core.getInput('name', { required: true });
 const extraPullNames = core.getInput('extraPullNames');
 const signingKey = core.getInput('signingKey');
 const authToken = core.getInput('authToken')
-const skipPush = core.getInput('skipPush');
+const skipPush = core.getBooleanInput('skipPush', { required: true });
```
Comment (sandydoo) [inline src/main.ts:17]: Do we still need `required: true` if there's a default?

**Review by sandydoo — CHANGES_REQUESTED** [review]

File: src/main.ts (hunk 2 above)
Comment (sestrella) [reply src/main.ts:17]: Good call; I think I misread the description here:

> If required set to be **false**, the input should have a default value in action.yml.

I read it as `true` the first time 😅 

https://github.com/actions/toolkit/tree/main/packages/core#inputsoutputs

**Review by sandydoo — APPROVED** [review]
Thanks! 🙏

**Comment by sandydoo** [comment]
@sestrella, could you please run `yarn build` and commit the changes? Then I can merge this.

**Comment by sestrella** [comment]
@sandydoo done!

**Pull request merged**